tests/framework/grid/GridViewTest.php

```diff
@@ -5,20 +5,63 @@
 
 use yii\data\ArrayDataProvider;
 use yii\grid\DataColumn;
-use yii\grid\GridView;
+use yiiunit\data\grid\TestGridView;
 
 /**
  * @author Evgeniy Tkachenko <et.coder@gmail.com>
  * @group grid
  */
 class GridViewTest extends \yiiunit\TestCase
 {
-    public function testGuessColumns()
+    protected function setUp()
     {
+        parent::setUp();
         $this->mockApplication();
+    }
+
+    /**
+     * @return array
+     */
+    public function emptyDataProvider()
+    {
+        return [
+            [null, 'No results found.'],
+            ['Empty', 'Empty'],
+            // https://github.com/yiisoft/yii2/issues/13352
+            [false, ''],
+        ];
+    }
+
+    /**
+     * @dataProvider emptyDataProvider
+     * @param mixed $emptyText
+     * @param string $expectedText
+     * @throws \Exception
+     */
+    public function testEmpty($emptyText, $expectedText)
+    {
+        $html = TestGridView::widget([
+            'dataProvider' => new ArrayDataProvider(['allModels' => []]),
+            'showHeader' => false,
+            'emptyText' => $emptyText,
+        ]);
+        $html = preg_replace("/\r|\n/", '', $html);
+
+        if ($expectedText) {
+            $emptyRowHtml = "<tr><td colspan=\"0\"><div class=\"empty\">{$expectedText}</div></td></tr>";
```

Comment:
`if` statements in tests are best to be prevented. It would be more readable without and in this case, directly in the dataProvider which will also improve the ease of comparison.